### PR TITLE
Update Notebooks to use new 3.4-ea2 artifacts (RHAB Images & RH-Index) 

### PR DESCRIPTION
**Pull request by atheo89**
Update Notebooks to use new 3.4-ea2 artifacts (RHAB Images & RH-Index) 

## Description

This PR looks huge but actually applies the following changes to all notebooks: 
- Update RH-Indexes to the new one 3.4-EA1 -> 3.4-EA2
- Update/align the the *.konflux.conf (rhoai) with *.conf (odh) to include also the rh-index there
- Update RHAB Images from 3.3* to 3.4-ea2* for downstream Dockerfiles, the ODH remain as they are
- Refresh the pylocks.toml everywhere to lock based the new index and codeserver .txt via `./scripts/lockfile-generators/create-requirements-lockfile.sh \
    --pyproject-toml codeserver/ubi9-python-3.12/pyproject.toml --download`
- Update couple of py packages as the old aren't available on the new rh-ndex (cve-updates)
- Fix the fallback index on pylock_generator so uv can select versions from fallback indexes when needed
- Update the dockerfile_fragments script with updated version uv==0.10.8 and apply it to all dockerfiles
- Update manifest imagestreams based the new package updates 

## How Has This Been Tested?

After a careful review of the image builds and tests, I can confirm that all the GHA builds succeed, while the Konflux builds fail due to an OOMKilled issue.

Regarding the tests, I observed several failures, all of which are already known issues:

ppc64le builds – Test pods never start and extend the timeouts ([link](https://github.com/opendatahub-io/notebooks/actions/runs/22893824281/job/66422986898?pr=3070#step:30:83))
Feast-related tests – Tests fail whenever Feast is used (datascience, tensorflow, pytorch, trusty, and all CUDA/ROCm variants). See the related issue: ([RHAIENG-3611](https://issues.redhat.com/browse/RHAIENG-3611)).
PyTorch tests – Fail because the onnxscript dependency is missing. This is a known issue: ([RHAIENG-3323](https://issues.redhat.com/browse/RHAIENG-3323)).
lllmcompressor tests – Fail due to an incorrect label name in the test configuration. Known test issue.

Self checklist (all need to be checked):
- [x] Ensure that you have run `make test` (`gmake` on macOS) before asking for review
- [x] Changes to everything except `Dockerfile.konflux` files should be done in `odh/notebooks` and automatically synced to `rhds/notebooks`. For Konflux-specific changes, modify `Dockerfile.konflux` files directly in `rhds/notebooks` as these require special attention in the downstream repository and flow to the upcoming RHOAI release.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

## Summary by CodeRabbit

* **Chores**
  * Switched package indexes from EA1 to EA2 across build configs and lockfiles.
  * Added/updated CPU fallback index entries and index-strategy for package resolution.
  * Upgraded the uv installer to 0.10.8 and refreshed multiple dependency versions.
  * Updated base image references in build arguments.

* **Tests**
  * Relaxed test expectation for a dependency version alignment.

### CODE REVIEW

File: jupyter/trustyai/ubi9-python-3.12/build-args/konflux.cpu.conf [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# Base Image   : RHEL 9.6 with Python 3.12
-# Architectures: linux/arm64, linux/ppc64le, linux/x86_64, linux/s360x
+INDEX_URL=https://console.redhat.com/api/pypi/public-rhai/rhoai/3.4-EA2/cpu-ubi9/simple/
 BASE_IMAGE=registry.redhat.io/rhai/base-image-cpu-rhel9:3.2.0-1764872006
 PYLOCK_FLAVOR=cpu
```
[the remark this answers is not in the document]
Comment (atheo89) [reply jupyter/trustyai/ubi9-python-3.12/build-args/konflux.cpu.conf]: fixed

File: runtimes/datascience/ubi9-python-3.12/build-args/cpu.conf [hunk 2]
```diff
@@ -1,3 +1,3 @@
-INDEX_URL=https://console.redhat.com/api/pypi/public-rhai/rhoai/3.4-EA1/cpu-ubi9/simple/
+INDEX_URL=https://console.redhat.com/api/pypi/public-rhai/rhoai/3.4-EA2/cpu-ubi9/simple/
 BASE_IMAGE=quay.io/opendatahub/odh-base-image-cpu-py312-c9s:latest
 PYLOCK_FLAVOR=cpu
```
[the remark this answers is not in the document]
Comment (atheo89) [reply runtimes/datascience/ubi9-python-3.12/build-args/cpu.conf]: This is the pattern for all this config files are related to ODH where we use different set of images

**Comment by ysok** [comment]
Most of Konflux failures are resource issue, OOMKilled, most of the image builds went out just fine. I've asked for their support in slack here https://redhat-internal.slack.com/archives/C02CTEB3MMF/p1773113789082019

**Comment by atheo89** [comment]
After a careful review of the image builds and tests, I can confirm that all the GHA builds succeed, while the Konflux builds fail due to an OOMKilled issue.

Regarding the tests, I observed several failures, all of which are already known issues:

- **ppc64le builds** – Test pods never start and extend the timeouts ([link](https://github.com/opendatahub-io/notebooks/actions/runs/22893824281/job/66422986898?pr=3070#step:30:83))
- **Feast-related tests** – Tests fail whenever Feast is used (datascience, tensorflow, pytorch, trusty, and all CUDA/ROCm variants). See the related issue: ([RHAIENG-3611](https://issues.redhat.com/browse/RHAIENG-3611)).
- **PyTorch tests** – Fail because the **onnxscript** dependency is missing. This is a known issue: ([RHAIENG-3323](https://issues.redhat.com/browse/RHAIENG-3323)).
- **lllmcompressor tests** – Fail due to an incorrect label name in the test configuration. Known test issue.

**Review by daniellutz — APPROVED** [review]
this is a big and really welcome update
glad to see the EA2 index coming up!
/lgtm
